packages/ws-server/src/metrics/wsMetricRegistry.ts

```diff
@@ -86,6 +91,50 @@ export default class WsMetricRegistry {
     });
   };
 
+  /**
+   * Generates a Prometheus Counter metric for tracking WebSocket method calls.
+   * Removes any existing metric with the same name from the provided registry before creating the new metric.
+   * @param {Registry} register - The Prometheus Registry where the metric will be registered.
+   * @param {any} gaugeInfo - Information of the counter.
+   * @returns {Gauge} Returns a new Counter metric instance.
+   */
+  private generateCpuGauge = (register: Registry, gaugeInfo: any): Gauge => {
```

Comment:
This function and the generateMemoryGauge() can be refactored for better code reusability. May be something like 

```suggestion
private generateCpuGauge = (register: Registry, gaugeInfo: any, collectFn?: any): Gauge => {
```
and down below some where make room for these particular collect functions?